pyteal/ast/gtxn.py

```diff
@@ -113,6 +137,113 @@ def group_index(cls, index):
     @classmethod
     def tx_id(cls, index):
         return cls(index, TxnField.tx_id)
-   
-    def type_of(self):
-        return type_of_field[self.field]
\ No newline at end of file
+
+    @classmethod
+    def application_id(cls, index):
+        return cls(index, TxnField.application_id)
+
+    @classmethod
+    def on_completion(cls, index):
+        return cls(index, TxnField.on_completion)
+
+    @classmethod
+    def approval_program(cls, index):
+        return cls(index, TxnField.approval_program)
+    
+    @classmethod
+    def clear_state_program(cls, index):
+        return cls(index, TxnField.clear_state_program)
+    
+    @classmethod
+    def rekey_to(cls, index):
+        return cls(index, TxnField.rekey_to)
+    
+    @classmethod
+    def config_asset(cls, index):
+        return cls(index, TxnField.config_asset)
+    
+    @classmethod
+    def config_asset_total(cls, index):
+        return cls(index, TxnField.config_asset_total)
+
+    @classmethod
+    def config_asset_decimals(cls, index):
+        return cls(index, TxnField.config_asset_decimals)
+    
+    @classmethod
+    def config_asset_default_frozen(cls, index):
+        return cls(index, TxnField.config_asset_default_frozen)
+    
+    @classmethod
+    def config_asset_unit_name(cls, index):
+        return cls(index, TxnField.config_asset_unit_name)
+
+    @classmethod
+    def config_asset_name(cls, index):
+        return cls(index, TxnField.config_asset_name)
+    
+    @classmethod
+    def config_asset_url(cls, index):
+        return cls(index, TxnField.config_asset_url)
+    
+    @classmethod
+    def config_asset_metadata_hash(cls, index):
+        return cls(index, TxnField.config_asset_metadata_hash)
+    
+    @classmethod
+    def config_asset_manager(cls, index):
+        return cls(index, TxnField.config_asset_manager)
+    
+    @classmethod
+    def config_asset_reserve(cls, index):
+        return cls(index, TxnField.config_asset_reserve)
+    
+    @classmethod
+    def config_asset_freeze(cls, index):
+        return cls(index, TxnField.config_asset_freeze)
+    
+    @classmethod
+    def config_asset_clawback(cls, index):
+        return cls(index, TxnField.config_asset_clawback)
+    
+    @classmethod
+    def freeze_asset(cls, index):
+        return cls(index, TxnField.freeze_asset)
+    
+    @classmethod
+    def freeze_asset_account(cls, index):
+        return cls(index, TxnField.freeze_asset_account)
+    
+    @classmethod
+    def freeze_asset_frozen(cls, index):
+        return cls(index, TxnField.freeze_asset_frozen)
+    
+    class ArrayAccessor:
+
+        def __init__(self, txIndex: int, accessField: TxnField, lengthField: TxnField) -> None:
+            self.txIndex = txIndex
+            self.accessField = accessField
+            self.lengthField = lengthField
+        
+        def length(self):
```

Comment:
Perhaps we could use `__len__` here? `https://docs.python.org/3/reference/datamodel.html#object.__len__`

Reply:
From that page:
> Should return the length of the object, an integer >= 0.

Unfortunately Python will raise an error if `__len__` does not return an integer, so this is the next best thing.